apis/Google.Cloud.Firestore/Google.Cloud.Firestore/FirestoreDb.cs

```diff
@@ -440,6 +445,10 @@ bool CheckRetry(RpcException e, ref bool rollback)
                         // No retry, but we do want to roll back.
                         rollback = true;
                         return false;
+                    // If the transaction has expired, don't roll back, but do retry.
+                    case StatusCode.InvalidArgument when e.Message?.IndexOf("transaction has expired", StringComparison.Ordinal) >= 0:
```

Comment:
Hope we could add a test upstream to make certain this text doesn't change in the future.

Reply:
Ideally we can add a machine-readable error description in the trailers :)
(But changing the message is certainly regarded as a breaking change.)